### PR TITLE
Add sample to enrich customer data

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: enrich-customer-data/main.bal [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+|};
+
+type Location record {
+    decimal lat;
+    decimal lng;
+};
+
+type Geometry record {
+    Location location;
+};
```
Comment (ShammiL) [inline enrich-customer-data/main.bal]: Can't we make these records closed records?

File: enrich-customer-data/Ballerina.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "lakshanweerasinghe"
```
Comment (SasinduDilshara) [inline enrich-customer-data/Ballerina.toml]: Shall we add a `org` name like ballerina

File: enrich-customer-data/main.bal [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    string status;
+};
+
+final http:Client geocodingClient = check new ("https://maps.googleapis.com");
+
+service /api/v1 on new http:Listener(8080) {
+    resource function get customerWithGeoCode(Customer customer) returns GeoTaggedCustomer|error {
+        // call the geocode api and retrive the lattiude and longitude information
+        GeocodeResponse response = check geocodingClient->
+            /maps/api/geocode/'json.get(address=customer.address, key=geocodingAPIKey);
```
Comment (SasinduDilshara) [inline enrich-customer-data/main.bal]:
```suggestion
            /maps/api/geocode/'json.get(address = customer.address, key = geocodingAPIKey);
```
WDYT?

File: enrich-customer-data/main.bal [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        if response.status == "OK" {
+            // enrich the customer data with location information
+            return {
+                ...customer,
+                lattitude: response.results[0].geometry.location.lat,
+                longitude: response.results[0].geometry.location.lng
+            };
+        }
+        return error("Error while getting the location information.");
```
Comment (SasinduDilshara) [inline enrich-customer-data/main.bal:47]: Shall we add this as a http Error Response like, `http:InternalServerError`. wdyt?

File: enrich-customer-data/main.bal (hunk 1 above)
Comment (LakshanWeerasinghe) [reply enrich-customer-data/main.bal]: The responses contains many unwanted fields for this specific sample.

File: enrich-customer-data/main.bal (hunk 4 above)
Comment (LakshanWeerasinghe) [reply enrich-customer-data/main.bal:47]: I think returning an error is fine. Because we anyway have a `check` expression in a previous statement.

**Review by SasinduDilshara — DISMISSED** [review]

File: enrich-customer-data/.devcontainer.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+    "image": "ballerina/ballerina-devcontainer:2201.8.2",
+    "extensions": ["WSO2.ballerina"],
+}
```
Comment (pcnfernando) [inline enrich-customer-data/.devcontainer.json]: Can remove this

File: enrich-customer-data/Ballerina.toml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "etl"
+name = "enrich_customer_data"
+version = "0.1.0"
+distribution = "2201.8.2"
+
+[build-options]
+observabilityIncluded = true
```
Comment (pcnfernando) [inline enrich-customer-data/Ballerina.toml]: Can remove this

File: enrich-customer-data/README.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+### Running the sample
+Run the following command from the root directory of the sample.
+```ballerina
+$ bal run
+```
+
+### Send a request
+```
+POST http://localhost:8080/api/v1/customerWithGeoCode
```
Comment (pcnfernando) [inline enrich-customer-data/README.md]: Have the curl command here

File: enrich-customer-data/main.bal [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    string status;
+};
+
+final http:Client geocodingClient = check new ("https://maps.googleapis.com");
+
+service /api/v1 on new http:Listener(8080) {
+    resource function post customerWithGeoCode(Customer customer) returns GeoTaggedCustomer|error {
+        // call the geocode api and retrive the lattiude and longitude information
+        GeocodeResponse response = check geocodingClient->
+            /maps/api/geocode/'json.get(address = customer.address, key = geocodingAPIKey);
```
Comment (pcnfernando) [inline enrich-customer-data/main.bal]:
```suggestion
            /maps/api/geocode/'json(address = customer.address, key = geocodingAPIKey);
```

Comment (pcnfernando) [reply enrich-customer-data/main.bal]: You don't need to say `get` explicitly

File: enrich-customer-data/main.bal [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    decimal lattitude;
+    decimal longitude;
+|};
+
+type Location record {
+    decimal lat;
+    decimal lng;
+};
+
+type Geometry record {
```
Comment (pcnfernando) [inline enrich-customer-data/main.bal]: Shall we nest the type definitions that only have/interested in one field

Comment (LakshanWeerasinghe) [reply enrich-customer-data/main.bal]: Please check the updated code.

File: enrich-customer-data/main.bal [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+configurable string geocodingAPIKey = ?;
+
+type Customer record {|
+    string customerId;
+    string address;
+|};
+
+type GeoTaggedCustomer record {|
+    *Customer;
+    decimal lattitude;
```
Comment (pcnfernando) [inline enrich-customer-data/main.bal]:
```suggestion
    decimal latitude;
```

File: enrich-customer-data/main.bal [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+};
+
+type GeocodeResponse record {
+    ResultsItem[] results;
+    string status;
+};
+
+final http:Client geocodingClient = check new ("https://maps.googleapis.com");
+
+service /api/v1 on new http:Listener(8080) {
```
Comment (pcnfernando) [inline enrich-customer-data/main.bal:34]: Make it isolated and the client readonly

**Review by pcnfernando — APPROVED** [review]
Suggestions will be handled separately. Hence approving

**Pull request merged**